### PR TITLE
move git and related functions from openshift.go into a separate file

### LINKED ISSUE #640: Move git functions in openshift.go into a separate package
We have some git functions like `GetGitCurrentRemoteURL()` and  `GetGitCurrentBranch()` in `opneshift.go`, it would be great if we move them to a separate package.

cc: @kadel

### PR DESCRIPTION
**Pull request by procrypt**
Fixes #640 and we can use the same functions in #518 
cc: @kadel @surajssd

### CODE REVIEW

**Comment by surajnarwade** [comment]
https://github.com/kubernetes-incubator/kompose/pull/655 will unblock this PR

**Comment by procrypt** [comment]
@surajnarwade Yes it will unblock it.

**Comment by cdrage** [comment]
@procrypt @surajnarwade 

Actually... there's a lot of things weird going on in this PR, weird indenting, changing of code to buildconfig. This all should just be a quick move and function name change, shouldn't it?

File: pkg/transformer/openshift/openshift.go [hunk 1]
```diff
@@ -449,48 +416,57 @@ func (o *OpenShift) Transform(komposeObject kobject.KomposeObject, opt kobject.C
...
+		}
 
-	// sort all object so all services are first
-	o.SortServicesFirst(&allobjects)
+		// If docker-compose has a volumes_from directive it will be handled here
+		o.VolumesFrom(&allobjects, komposeObject)
 
+		// sort all object so all services are first
+		o.SortServicesFirst(&allobjects)
+	}
```
Comment (cdrage) [inline pkg/transformer/openshift/openshift.go]: move weird changes

File: pkg/transformer/openshift/openshift.go [hunk 2]
```diff
@@ -449,48 +416,57 @@ func (o *OpenShift) Transform(komposeObject kobject.KomposeObject, opt kobject.C
...
-	o.VolumesFrom(&allobjects, komposeObject)
+			// Update and then append the objects (we're done generating)
+			o.UpdateKubernetesObjects(name, service, &objects)
+			allobjects = append(allobjects, objects...)
+		}
 
-	// sort all object so all services are first
-	o.SortServicesFirst(&allobjects)
+		// If docker-compose has a volumes_from directive it will be handled here
+		o.VolumesFrom(&allobjects, komposeObject)
```
Comment (cdrage) [inline pkg/transformer/openshift/openshift.go]: weird changes

File: pkg/transformer/openshift/openshift.go [hunk 3]
```diff
@@ -449,48 +416,57 @@ func (o *OpenShift) Transform(komposeObject kobject.KomposeObject, opt kobject.C
...
+					objects = append(objects, svc)
 
-				if service.ExposeService != "" {
-					objects = append(objects, o.initRoute(name, service, svc.Spec.Ports[0].Port))
+					if service.ExposeService != "" {
+						objects = append(objects, o.initRoute(name, service, svc.Spec.Ports[0].Port))
+					}
+				} else {
+					svc := o.CreateHeadlessService(name, service, objects)
+					objects = append(objects, svc)
```
Comment (cdrage) [inline pkg/transformer/openshift/openshift.go]: again, changes to unrelated code

File: pkg/transformer/openshift/openshift.go [hunk 4]
```diff
@@ -449,48 +416,57 @@ func (o *OpenShift) Transform(komposeObject kobject.KomposeObject, opt kobject.C
...
+				}
 
-			// If ports not provided in configuration we will not make service
-			if o.PortsExist(name, service) {
-				svc := o.CreateService(name, service, objects)
-				objects = append(objects, svc)
+				// If ports not provided in configuration we will not make service
+				if o.PortsExist(name, service) {
+					svc := o.CreateService(name, service, objects)
+					objects = append(objects, svc)
```
Comment (cdrage) [inline pkg/transformer/openshift/openshift.go]: weird indenting?

File: pkg/transformer/openshift/openshift.go [hunk 5]
```diff
@@ -449,48 +416,57 @@ func (o *OpenShift) Transform(komposeObject kobject.KomposeObject, opt kobject.C
...
+
+						if !git.HasGitBinary() && (buildRepo == "" || buildBranch == "") {
+							return nil, errors.New("Git is not installed! Please install Git to create buildconfig, else supply source repository and branch to use for build using '--build-repo', '--build-branch' options respectively")
+						}
+						if buildBranch == "" {
+							buildBranch, err = git.GetGitCurrentBranch(composeFileDir)
+							if err != nil {
+								return nil, errors.Wrap(err, "Buildconfig cannot be created because current git branch couldn't be detected.")
+							}
+						}
```
Comment (cdrage) [inline pkg/transformer/openshift/openshift.go]: weird indenting again, you indented all of this code.

**Comment by surajnarwade** [comment]
@procrypt run gofmt

**Comment by surajnarwade** [comment]
@procrypt @cdrage , I think as `git` related functions are only required by openshift, `git.go` should be in `pkg/transformer/openshift/git.go`,
thoughts ?

**Comment by cdrage** [comment]
@surajnarwade makes sense to put in different folder even though only OpenShift uses it.. we may use it for Kubernetes in the future?

**Comment by procrypt** [comment]
@surajnarwade we will be using it for Functional Tests also.

**Comment by cdrage** [comment]
LGTM

**Comment by procrypt** [comment]
Ping @kadel Please review.

**Comment by kadel** [comment]
Should those functions be used in openshift code for buildconfig stuff etc...?

**Comment by surajnarwade** [comment]
@kadel , yeah buildconfig stuff

**Comment by cdrage** [comment]
@procrypt Mind fixing the conflicting file and then we can do another review / possible merge?

**Comment by surajnarwade** [comment]
@cdrage @kadel , I have updated @procrypt PR, good to go now :)

**Comment by surajssd** [comment]
@surajnarwade the PR says that code is being moved, but I don't see code being removed from somewhere else, it's just code addition? can you please elaborate what is happening?

**Comment by surajnarwade** [comment]
@cdrage @surajssd @kadel needs your review here

**Comment by cdrage** [comment]
This LGTM.

**Comment by cdrage** [comment]
@surajssd ?
